File: app/src/main/java/com/readtracker/android/adapters/GoogleBookItem.java
package com.readtracker.android.adapters;

import com.readtracker.android.support.GoogleBook;

/**
 * Shows a GoogleBook search result
 */
public class GoogleBookItem extends BookItem {
  public GoogleBookItem(String title, String author) {
    super(title, author, null);
  }

  public GoogleBookItem(GoogleBook googleBook) {
    this(googleBook.getTitle(), googleBook.getAuthor());
    coverURL = googleBook.getCoverURL();
    this.pageCount = googleBook.getPageCount();
  }
}

File: app/src/main/java/com/readtracker/android/adapters/GoogleBookItem.kt
package com.readtracker.android.adapters

import com.readtracker.android.support.GoogleBooksSearchService

/**
 * Shows a GoogleBook search result
 */
data class GoogleBookItem(
    override val title: String,
    override val author: String,
    override val coverUrl: String,
    override val pageCount: Long
) : BookItem(title, author, coverUrl, pageCount)
